### PR TITLE
GH-6570 Add Friedman and Popescu's H statistics doc (GBM, XGBoost)

### PR DESCRIPTION
**Pull request by maurever**
Issue: #6570

### CODE REVIEW

**Comment by maurever** [comment]
@hannah-tillman @wendycwong, I added a new section to the GBM and XGBoost section. I am not sure if it is okay, or if we have a better place to document it.

File: h2o-docs/src/product/data-science/gbm.rst [hunk 1]
```diff
@@ -358,6 +358,21 @@ Metrics
...
+GBM Friedman and Popescu's H statistics
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+You can calculates the Friedman and Popescu's H statistics to test for the presence of an interaction between specified variables. 
+
+H varies from 0 to 1. It will have a value of 0 if the model exhibits no interaction between specified variables and a correspondingly larger value for a stronger interaction effect between them. NaN is returned if a computation is spoiled by weak main effects and rounding errors.
+
+This statistic can only be calculated for numerical variables. Missing values are supported. 
+
+See Jerome H. Friedman and Bogdan E. Popescu, 2008, "Predictive learning via rule ensembles", *Ann. Appl. Stat.* **2**:916-954, http://projecteuclid.org/download/pdfview_1/euclid.aoas/1223908046, s. 8.1.
```
Comment (hannah-tillman) [inline h2o-docs/src/product/data-science/gbm.rst:370]: Can this be added to the reference section on the algo page or does it need to be in this section??

File: h2o-docs/src/product/data-science/xgboost.rst [hunk 2]
```diff
@@ -373,6 +373,21 @@ Metrics
...
+XGBoost Friedman and Popescu's H statistics
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+You can calculates the Friedman and Popescu's H statistics to test for the presence of an interaction between specified variables. 
+
+H varies from 0 to 1. It will have a value of 0 if the model exhibits no interaction between specified variables and a correspondingly larger value for a stronger interaction effect between them. NaN is returned if a computation is spoiled by weak main effects and rounding errors.
+
+This statistic can only be calculated for numerical variables. Missing values are supported. 
+
+See Jerome H. Friedman and Bogdan E. Popescu, 2008, "Predictive learning via rule ensembles", *Ann. Appl. Stat.* **2**:916-954, http://projecteuclid.org/download/pdfview_1/euclid.aoas/1223908046, s. 8.1.
```
Comment (hannah-tillman) [inline h2o-docs/src/product/data-science/xgboost.rst:385]: Same w/ this one?

**Comment by wendycwong** [comment]
I have no problem with the writing.  @hannah-tillman :  Please approve after resolving all issues.

File: h2o-docs/src/product/data-science/gbm.rst (hunk 1 above)
Comment (maurever) [reply h2o-docs/src/product/data-science/gbm.rst]: Done. Thanks!

File: h2o-docs/src/product/data-science/xgboost.rst (hunk 2 above)
Comment (maurever) [reply h2o-docs/src/product/data-science/xgboost.rst]: Done. Thanks!

**Review by hannah-tillman — APPROVED** [review]
LGTM -- thanks :)

**Pull request merged**